pkg/utils/vault/client.go

```diff
@@ -23,10 +24,25 @@ const (
 	ghTokenEnvVar = "GITHUB_TOKEN" //nolint:gosec
 )
 
+var once sync.Once
```

Comment:
with the `sync.Once` at the package level and the variable that holds the client in a closure of the `NewClientProvider` won't you be able to generate a `nil` client if two pieces of code call `NewClientProvider`? So I would expect the `once` also to be in the closure or the `client` variable also be global. Or am I misremembering how the `sync` package works (totally possible)?

Reply:
You are completely right! 🤦